File: proton-sdk/incubating/client/kt/sdk/src/main/kotlin/me/proton/drive/sdk/telemetry/DownloadEvent.kt
package me.proton.drive.sdk.telemetry

data class DownloadEvent(
    val volumeType: VolumeType,
    val claimedFileSize: Long,
    val approximateClaimedFileSize: Long,
    val downloadedSize: Long,
    val approximateDownloadedSize: Long,
    val error: DownloadError? = null,
    val originalError: String? = null,
)

File: proton-sdk/client/cs/src/Proton.Drive.Sdk/Telemetry/DownloadEvent.cs
using System.Text.Json.Serialization;
using Proton.Sdk.Telemetry;

namespace Proton.Drive.Sdk.Telemetry;

public sealed class DownloadEvent : IMetricEvent
{
    public string Name => "download";

    public required VolumeType VolumeType { get; init; }

    public long DownloadedSize { get; set; }

    public long ApproximateDownloadedSize { get; set; }

    public long? ClaimedFileSize { get; set; }

    public long? ApproximateClaimedFileSize { get; set; }

    public DownloadError? Error { get; set; }

    [JsonIgnore]
    public Exception? OriginalError { get; set; }
}
